src/NuGet.Core/NuGet.ProjectModel/Utf8JsonStreamPackageSpecReader.cs

```diff
@@ -0,0 +1,1873 @@
+                            ReferenceType = LibraryDependencyReferenceType.Transitive
+                        };
+
+                        results.Add(libraryDependency);
+                    }
+                }
+            }
+        }
+
+        private static void ReadDependencies(
+            ref Utf8JsonStreamReader jsonReader,
+            IList<LibraryDependency> results,
+            string packageSpecPath,
+            bool isGacOrFrameworkReference)
+        {
+            if (jsonReader.Read() && jsonReader.TokenType == JsonTokenType.StartObject)
+            {
+                while (jsonReader.Read() && jsonReader.TokenType == JsonTokenType.PropertyName)
+                {
+                    var propertyName = jsonReader.GetString();
+                    if (string.IsNullOrEmpty(propertyName))
+                    {
+                        throw FileFormatException.Create("Unable to resolve dependency ''.", packageSpecPath);
+                    }
+
+                    // Support
+                    // "dependencies" : {
+                    //    "Name" : "1.0"
+                    // }
+
+                    if (jsonReader.Read())
+                    {
+                        var dependencyIncludeFlagsValue = LibraryIncludeFlags.All;
+                        var dependencyExcludeFlagsValue = LibraryIncludeFlags.None;
+                        var suppressParentFlagsValue = LibraryIncludeFlagUtils.DefaultSuppressParent;
+                        List<NuGetLogCode> noWarn = null;
+
+                        // This method handles both the dependencies and framework assembly sections.
+                        // Framework references should be limited to references.
+                        // Dependencies should allow everything but framework references.
+                        LibraryDependencyTarget targetFlagsValue = isGacOrFrameworkReference
+                            ? LibraryDependencyTarget.Reference
+                            : LibraryDependencyTarget.All & ~LibraryDependencyTarget.Reference;
+
+                        var autoReferenced = false;
+                        var generatePathProperty = false;
+                        var versionCentrallyManaged = false;
+                        string aliases = null;
+                        string dependencyVersionValue = null;
+                        VersionRange versionOverride = null;
+
+                        if (jsonReader.TokenType == JsonTokenType.String)
+                        {
+                            dependencyVersionValue = jsonReader.GetString();
+                        }
+                        else if (jsonReader.TokenType == JsonTokenType.StartObject)
+                        {
+                            while (jsonReader.Read() && jsonReader.TokenType == JsonTokenType.PropertyName)
+                            {
+                                IEnumerable<string> values = null;
```

Comment:
Same as earlier.

Reply:
There's a bunch of cleanup that the editor suggests that could make these changes shorter: 
![image](https://github.com/NuGet/NuGet.Client/assets/2878341/bfb20020-4a3f-44d0-9207-075fcf881d91)
.